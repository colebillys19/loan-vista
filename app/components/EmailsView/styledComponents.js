import styled from 'styled-components';

export const EmailsViewWrapper = styled.div``;
