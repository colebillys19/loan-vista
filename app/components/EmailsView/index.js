/**
 * EmailsView
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import EmailsList from 'components/EmailsList';
import TabWrapper from 'components/TabWrapper';
import ConditionalRender from 'components/_baseUI/ConditionalRender';
import Spinner from 'components/_baseUI/Spinner';
import NoListDataFallback from 'components/_baseUI/NoListDataFallback';

import { EmailsViewWrapper } from './styledComponents';

const EmailsView = ({ emailsData, loading }) => {
  const noListData = !loading && emailsData.length === 0;

  return (
    <TabWrapper
      aria-labelledby="emails-and-letters-tab"
      id="emails-and-letters-view"
    >
      <EmailsViewWrapper>
        <ConditionalRender
          Component={
            <ConditionalRender
              Component={
                <EmailsList emailsData={emailsData} loading={loading} />
              }
              FallbackComponent={<Spinner />}
              shouldRender={!loading}
            />
          }
          FallbackComponent={<NoListDataFallback />}
          shouldRender={!noListData}
        />
      </EmailsViewWrapper>
    </TabWrapper>
  );
};

EmailsView.propTypes = {
  emailsData: T.array.isRequired,
  loading: T.bool.isRequired,
};

export default EmailsView;
