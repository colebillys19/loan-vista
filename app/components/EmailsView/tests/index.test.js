import React from 'react';
import { render } from 'react-testing-library';

import EmailsView from '../index';

describe('<EmailsView />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<EmailsView />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<EmailsView />);
    expect(firstChild).toMatchSnapshot();
  });
});
