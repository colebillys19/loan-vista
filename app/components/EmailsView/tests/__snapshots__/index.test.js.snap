// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailsView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="emails-tab"
  class="TabWrapper-j2wsfb-0 beIMbY"
  id="emails-view"
>
  <div
    class="styledComponents__EmailsViewWrapper-zcwy7r-0 iHmmMF"
  >
    EmailsView
  </div>
</section>
`;
